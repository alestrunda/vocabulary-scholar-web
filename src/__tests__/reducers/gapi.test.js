import gapiReducer from '../../reducers/gapi';
import {
  gapiClientStatusUpdate,
  gapiConnected,
  gapiSetStoreFileID,
  gapiStoreLoadedFromDrive,
} from '../../actions/gapi';

it('correct gapi initial state', () => {
  const action = { type: '' }; //passing no action should return initial state
  const initialState = {};
  const defaultState = undefined;

  expect(gapiReducer(defaultState, action)).toEqual(initialState);
});

it('handles GAPI_CLIENT_STATUS_UPDATE', () => {
  const isClientSignedIn = true,
    isInitial = true;
  const action = gapiClientStatusUpdate(isClientSignedIn, isInitial);
  const defaultState = {};

  expect(gapiReducer(defaultState, action)).toMatchSnapshot();
});

it('handles GAPI_CONNECTED', () => {
  const action = gapiConnected();
  const expectedState = {
    isConnected: true,
  };
  const defaultState = {};

  expect(gapiReducer(defaultState, action)).toEqual(expectedState);
});

it('handles GAPI_SET_STORE_FILE_ID', () => {
  const storeFileID = 'abc123';
  const action = gapiSetStoreFileID(storeFileID);
  const defaultState = {};

  expect(gapiReducer(defaultState, action)).toMatchSnapshot();
});

it('handles GAPI_STORE_LOADED_FROM_DRIVE', () => {
  const isStoreLoadedFromDrive = true,
    storeFileID = 'abc123';
  const action = gapiStoreLoadedFromDrive(isStoreLoadedFromDrive, storeFileID);
  const defaultState = {};

  expect(gapiReducer(defaultState, action)).toMatchSnapshot();
});
