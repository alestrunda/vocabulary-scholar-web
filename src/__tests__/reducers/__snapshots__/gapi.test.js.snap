// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles GAPI_CLIENT_STATUS_UPDATE 1`] = `
Object {
  "isClientSignedIn": true,
  "isStoreLoadedFromDrive": undefined,
}
`;

exports[`handles GAPI_SET_STORE_FILE_ID 1`] = `
Object {
  "storeFileID": "abc123",
}
`;

exports[`handles GAPI_STORE_LOADED_FROM_DRIVE 1`] = `
Object {
  "isStoreLoadedFromDrive": true,
  "storeFileID": "abc123",
}
`;
